packages/devtools_app/lib/src/screens/memory/panes/chart/controller/chart_connection.dart

```diff
@@ -35,36 +34,64 @@ class ChartConnection extends DisposableController
     isAndroidChartVisible: isAndroidChartVisible,
   );
 
-  Timer? _pollingTimer;
-  bool _connected = false;
+  bool initialized = false;
+
+  DebounceTimer? _polling;
+
+  late final bool isDeviceAndroid;
+
+  void maybeInit() async {
+    if (initialized) return;
+    initialized = true;
```

Comment:
put this at the end of the method so we are setting it to true after all the initialization has occurred.

Reply:
Then we will have to remember to do it in every early return. Moved.